Puzzlepart/prosjektportalen365
# Bundle 'Idea addon' directly

### Your checklist for this pull request


### Description

_Draft PR: changes and additions are subject to change_

## Changes:

Fields with changed internal names also has new GUID's. These are added to the PostInstallUpgrade.ps1 script and data will be merged correctly.

- Fields (Idebehandling)
  - GtIdeaExpectedGain => `GtIdeaExpectedGains`
  - GtIdeaExecutionResourceNeeds => `GtIdeaExecutionPlanResourceNeeds`
  - GtIdeaExecutionSuccessFactors => `GtIdeaExecutionPlanSuccessFactors`
  - New Choice for `GtIdeaDecision`
- Commandbar icons changed
- ListViewCommand
  - ~~Added better handling of approved idea registrations and processing~~ (scrapped for now, todo in a later issue)
  - ~~Added information dialog for better information regarding what will happen when users approves/rejects registrations/processing elements~~  (scrapped for now, todo in a later issue)

- ProjectInformation webpart
  - Added prop to turn on use of Ideaprocessing to be able to sync properties from the new list `Prosjektdata` with a callout dialog for more information about what this does (see images)
  - New button `Synkroniser prosjektegenskaper`
  - New dialog for synchronizing properties

![image](https://user-images.githubusercontent.com/28678468/193766046-683ab0c5-1ae6-4dd6-8e89-2d8f57b3423f.png)


## Additions:

- New SharePointFramework package: `PortfolioExtensions`
- New npm package: `pp365-portfolioExtensions`
- Fields
  - CustomFormatting for fields have been added directly with the provisioning
    - `GtIdeaUrl`, `GtIdeaRecommendation` and `GtIdeaDecision`
- Lists
  - `Prosjektdata`
    - This list has most of the SiteFields found in `Prosjekter` list
    - List is used to syncronized project properties to projects 
- ListViewCommand
  - `Opprett prosjektdata` command has been added to Idébehandling list
    - This created a new element in `Prosjektdata` list, connects this item to the element selected in `Idébehandling` and then opens an EditForm for the new element.

### How to test

1. Create and register idea, approve this
2. Fill out element in 'Idébehandlingsliste', approve this
3. Create  'Prosjektdata' through the dialog for the and approved element
4. Fill out SiteUrl for 'Prosjektdata' element
5. Go to the project where 'useIdeaProcessing' is activated
6. Press button: 'Synkroniser prosjektegenskaper'
7. See that the values from 'Prosjektdata' are synchronized to 'Prosjektegenskaper' and afterwards synchronized to portfoliolist 'Prosjekter'

### Relevant issues (if applicable)

#777
